### PR TITLE
fix(install): fix 9915

### LINKED ISSUE #9915: `helm install` installs crds from disabled child charts
https://github.com/helm/helm/blob/main/pkg/chart/chart.go#L153-L168

We have a parent chart that wants to conditionally enable/disable child charts. We don't want crds from child charts to be installed in the cluster, that child chart is not enabled. From Helm code, it seems that conditions are not considered when calculating `CRDObjects()`. This feels like a bug.

Output of `helm version`: v3.6.1

Output of `kubectl version`:

Cloud Provider/Platform (AKS, GKE, Minikube etc.):

### PR DESCRIPTION
**Pull request by cndoit18**
Signed-off-by: cndoit18 <cndoit18@outlook.com>

**What this PR does / why we need it**:

fixes: #9915 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by yxxhero** [comment]
@cndoit18 from the issue, it  conditionally enable/disable child charts. example: 
```yaml
dependencies: # A list of the chart requirements (optional)
  - name: The name of the chart (nginx)
    version: The version of the chart ("1.2.3")
    repository: (optional) The repository URL ("https://example.com/charts") or alias ("@repo-name")
    condition: (optional) A yaml path that resolves to a boolean, used for enabling/disabling charts (e.g. subchart1.enabled )
```
I think you should decide whether to load chart's crds by **condition** filed. do you think of?

**Comment by cndoit18** [comment]
> @cndoit18 from the issue, it conditionally enable/disable child charts. example:
> 
> ```yaml
> dependencies: # A list of the chart requirements (optional)
>   - name: The name of the chart (nginx)
>     version: The version of the chart ("1.2.3")
>     repository: (optional) The repository URL ("https://example.com/charts") or alias ("@repo-name")
>     condition: (optional) A yaml path that resolves to a boolean, used for enabling/disabling charts (e.g. subchart1.enabled )
> ```
> 
> I think you should decide whether to load chart's crds by **condition** filed. do you think of?

Very good, I gave missed this part. I'll rethink it.

File: pkg/action/action_test.go [hunk 1]
```diff
@@ -289,3 +290,12 @@ func TestGetVersionSet(t *testing.T) {
 		t.Error("Non-existent version is reported found.")
 	}
 }
+
+func loadChart(t *testing.T, path string) *chart.Chart {
```
Comment (yxxhero) [inline pkg/action/action_test.go:294]: this is a utils func, why has testing.T arg? you can return (*chart.Chart, error), WDYT?

**Review by yxxhero — CHANGES_REQUESTED** [review]

**Review by yxxhero — APPROVED** [review]
LGTM
@bacongobbler what do  you think of?

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @cndoit18 for the fix. Verified by testing manually with and without the fix.  I cannot think of any way we can test this with CRDs, so unit test is not required.

This requires another core maintainer to merge.

**Review by bacongobbler — APPROVED** [review]

**Pull request merged**